src/Common/Internal/ServiceRestProxy.php

```diff
@@ -50,78 +50,86 @@ class ServiceRestProxy extends RestProxy
      * @var string
      */
     private $_accountName;
-    
+
     /**
-     * 
+     *
      * @var \Uri
      */
        private $_psrUri;
+
+    /**
+     * @var array
+     */
+    private $guzzleOptions;
     
     /**
      * Initializes new ServiceRestProxy object.
      *
      * @param string      $uri            The storage account uri.
      * @param string      $accountName    The name of the account.
      * @param ISerializer $dataSerializer The data serializer.
+     * @param array       $guzzleOptions  Array of options to pass to the Guzzle client
      */
-    public function __construct($uri, $accountName, $dataSerializer)
-    {           
+    public function __construct($uri, $accountName, $dataSerializer, $guzzleOptions = [])
```

Comment:
how about name it 'options' and use 'guzzle' => [] element inside options for guzzle specific settings? Then we can add more general options in the further without adding more parameters.